app/Main.hs

```diff
@@ -178,6 +196,16 @@ showLogForTask (RunningTask _ _ _ _ p) =
         _ <- logLoop mout mVar
         terminateProcess ph
 
+killTask :: String -> ExceptT String ApplicationContext ()
+killTask = findTaskRun $ \t -> lift $ do
+  code <-
+    terminateProcess (rtPh t)
```

Comment:
Instead of terminating the process, I would really like it if we sent a SIGINT, instead of SIGTERM to the process then wait for it to exit. Just so that the child process can also do its respective cleanup if it has a handler for SIGINT.